### PR TITLE
Check backward compatibility of metadata command deserialization

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:2ea89d24

### CODE REVIEW

File: scarb-metadata/tests/command.rs [hunk 1]
```diff
@@ -1,14 +1,22 @@
 use assert_fs::TempDir;
 use snapbox::cmd::{cargo_bin, Command};
+use std::env;
+use std::path::PathBuf;
 
 use scarb_metadata::MetadataCommand;
 
+fn scarb_bin() -> PathBuf {
+    env::var_os("SCARB_BIN_PATH")
```
Comment (mkaput) [inline scarb-metadata/tests/command.rs:9]:
```suggestion
    env::var_os("SCARB")
```

let's use our standard var name?

File: scarb-metadata/tests/command.rs [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+        .map(PathBuf::from)
+        .unwrap_or(cargo_bin("scarb"))
```
Comment (mkaput) [inline scarb-metadata/tests/command.rs:11]:
```suggestion
        .unwrap_or_else(|| cargo_bin("scarb"))
```

File: scarb-metadata/tests/command.rs (hunk 1 above)
Comment (maciektr) [reply scarb-metadata/tests/command.rs:9]: I have used a different var on purpose, not to confuse with other configuration when running tests locally on user env.

Comment (mkaput) [reply scarb-metadata/tests/command.rs:9]: Hmm, this was not obvious from me from `SCARB_BIN_PATH` name alone 😕 Your reasoning seems right, so perhaps rename this to something obviously telling that this influences tests?

**Review by mkaput — APPROVED** [review]

**Pull request merged**